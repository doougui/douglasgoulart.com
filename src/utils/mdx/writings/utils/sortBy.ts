import { Writing } from 'types/Writing';
import { parseToDate } from 'utils/time/parseToDate';
import { SortTypesKeys } from '../types';

function byNew(data: Writing[]) {
  return data.sort((a, b) => {
    const aAsDate = parseToDate(a.publishedAt);
    const bAsDate = parseToDate(b.publishedAt);
    return Number(bAsDate) - Number(aAsDate);
  });
}

function byName(data: Writing[]) {
  return data.sort((a, b) => (a.title > b.title ? 1 : -1));
}

export function sortBy(data: Writing[], type: SortTypesKeys): Writing[] {
  const sortTypes: { [key in SortTypesKeys]: () => Writing[] } = {
    new: () => byNew(data),
    name: () => byName(data),
  };

  return sortTypes[type]();
}
