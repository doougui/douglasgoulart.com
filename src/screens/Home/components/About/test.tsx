import { render, screen } from 'utils/tests';
import { About } from '.';

describe('<About />', () => {
  it('should render correctly', () => {
    render(<About />);

    expect(
      screen.getByText(
        'I‘m a UI developer based in Barcelona. I‘m interested in design systems, web3, user/dev experience and under engineering.',
      ),
    ).toBeInTheDocument();

    expect(
      screen.getByText(
        'I‘m currently doing UI/DX at Rainbow to help create the world‘s friendliest wallet.',
      ),
    ).toBeInTheDocument();

    expect(
      screen.getByText(
        'Previously, I worked at Modulz. Where I co-created Radix, designed the Stitches API and led the developer community.',
      ),
    ).toBeInTheDocument();
  });

  it('should render "More about me" link and social media links', () => {
    render(<About />);

    expect(screen.getByRole('link', { name: 'More about me' })).toHaveAttribute(
      'href',
      '/about',
    );

    expect(screen.getByTestId('github-link')).toHaveAttribute(
      'href',
      'https://github.com/doougui',
    );

    expect(screen.getByTestId('linkedin-link')).toHaveAttribute(
      'href',
      'https://www.linkedin.com/in/douglaspigoulart/',
    );

    expect(screen.getByTestId('twitter-link')).toHaveAttribute(
      'href',
      'https://twitter.com/oDougui',
    );
  });
});
