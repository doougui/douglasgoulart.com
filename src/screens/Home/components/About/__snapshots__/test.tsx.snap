// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> should render correctly 1`] = `
.c2 {
  color: #107BEB;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
}

.c0 {
  line-height: 1.8rem;
}

.c0 p {
  font-size: 1.6rem;
  line-height: 2.4rem;
  margin-bottom: 3rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1rem;
}

<section
  class="c0"
>
  <p>
    Hi, my name is Douglas. I'm a software developer based in Brazil that likes to play some games and write about some cool stuff I learn along my journey as a software dev.
  </p>
  <p>
    Iâ€˜m currently doing front-end and mobile development at Tecimob where I managed to generate value and ease the life of more than three thousand real estate agencies whilst improving the quality of existing software and creating new, valuable software from the beginning.
  </p>
  <p>
    Most of my experience comes down to working with JavaScript, TypeScript, React, React Native and Next.js.
  </p>
  <div
    class="c1"
  >
    <a
      class="c2"
      href="/about"
    >
      More about me
    </a>
    <div
      class="c3"
    >
      <a
        aria-label="Douglas' GitHub"
        href="https://github.com/doougui"
        target="_blank"
      >
        <svg
          color="#888888"
          fill="currentColor"
          height="24"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(136, 136, 136);"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 2C6.475 2 2 6.475 2 12a9.994 9.994 0 0 0 6.838 9.488c.5.087.687-.213.687-.476 0-.237-.013-1.024-.013-1.862-2.512.463-3.162-.612-3.362-1.175-.113-.288-.6-1.175-1.025-1.413-.35-.187-.85-.65-.013-.662.788-.013 1.35.725 1.538 1.025.9 1.512 2.338 1.087 2.912.825.088-.65.35-1.087.638-1.337-2.225-.25-4.55-1.113-4.55-4.938 0-1.088.387-1.987 1.025-2.688-.1-.25-.45-1.275.1-2.65 0 0 .837-.262 2.75 1.026a9.28 9.28 0 0 1 2.5-.338c.85 0 1.7.112 2.5.337 1.912-1.3 2.75-1.024 2.75-1.024.55 1.375.2 2.4.1 2.65.637.7 1.025 1.587 1.025 2.687 0 3.838-2.337 4.688-4.562 4.938.362.312.675.912.675 1.85 0 1.337-.013 2.412-.013 2.75 0 .262.188.574.688.474A10.016 10.016 0 0 0 22 12c0-5.525-4.475-10-10-10z"
            />
          </g>
        </svg>
      </a>
      <a
        aria-label="Douglas' Linkedin"
        href="https://www.linkedin.com/in/douglaspigoulart/"
        target="_blank"
      >
        <svg
          color="#888888"
          fill="currentColor"
          height="24"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(136, 136, 136);"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M18.335 18.339H15.67v-4.177c0-.996-.02-2.278-1.39-2.278-1.389 0-1.601 1.084-1.601 2.205v4.25h-2.666V9.75h2.56v1.17h.035c.358-.674 1.228-1.387 2.528-1.387 2.7 0 3.2 1.778 3.2 4.091v4.715zM7.003 8.575a1.546 1.546 0 0 1-1.548-1.549 1.548 1.548 0 1 1 1.547 1.549zm1.336 9.764H5.666V9.75H8.34v8.589zM19.67 3H4.329C3.593 3 3 3.58 3 4.297v15.406C3 20.42 3.594 21 4.328 21h15.338C20.4 21 21 20.42 21 19.703V4.297C21 3.58 20.4 3 19.666 3h.003z"
              fill-rule="nonzero"
            />
          </g>
        </svg>
      </a>
      <a
        aria-label="Douglas' E-mail"
        href="mailto:douglaspigoulart@gmail.com"
        target="_blank"
      >
        <svg
          color="#888888"
          fill="currentColor"
          height="24"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(136, 136, 136);"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3 3h18a1 1 0 0 1 1 1v16a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V4a1 1 0 0 1 1-1zm9.06 8.683L5.648 6.238 4.353 7.762l7.72 6.555 7.581-6.56-1.308-1.513-6.285 5.439z"
            />
          </g>
        </svg>
      </a>
    </div>
  </div>
</section>
`;
